core/utils/utils.go

```diff
@@ -487,85 +487,84 @@ func (da *dependentAwaiter) DependentReady() {
 }
 
 // BoundedQueue is a FIFO queue that discards older items when it reaches its capacity.
-type BoundedQueue struct {
-	capacity uint
-	items    []interface{}
-	mu       *sync.RWMutex
+type BoundedQueue[T any] struct {
```

Comment:
Just noticed that `BoundedQueue` and `Mailbox` are pretty much the same - both are bounded FIFO queues that end up dropping older items once they reach capacity, except that `Mailbox` has `chNotify`. Not sure if it's worth keeping BoundedQueue around?

Reply:
API's are slightly different but perhaps they could be unified. Mailbox seems to have a superset of functionality at a glance.

Reply:
SGTM, though maybe better for a follow up to keep this PR simple.

Reply:
Can we have a ticket to track?